tests/main/classic-custom-device-reg/task.yaml

```diff
@@ -15,7 +15,8 @@ environment:
     SEED_DIR: /var/lib/snapd/seed
 
 prepare: |
-    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+    # In this scenario, the keys from the snapd pkg are used
+    if [ "$TRUST_TEST_KEYS" = "false" ] || tests.info is-snapd-pkg-repo; then
```

Comment:
I was thinking if this is really the case. Snapd classic will re-exec into snapd snap that may be built with test keys. Does it really fail?

Reply:
there are some scenarios where we don't have snapd snap installed, so it uses the keys from the deb. 
I'll add a comment there

Reply:
Thanks, this sounds good. If We identify all the cases that don't have snapd snap installed then we can work on fixing them. I assume that the point of the test was not to test that specific behavior and that we can fix them over time.

Reply:
right, my idea is to create a following pr to cleanup some scenarios and fix other tests